main.tf

```diff
@@ -48,15 +50,63 @@ resource "ibm_resource_instance" "appid" {
   location          = var.region
   resource_group_id = var.resource_group_id
   tags              = var.resource_tags
-  parameters = {
+  parameters = local.parameters_enabled ? {
     kms_info = "{\"id\": \"${var.existing_kms_instance_guid}\"}"
     tek_id   = var.kms_key_crn
-  }
+  } : null
 }
 
 resource "ibm_resource_key" "resource_keys" {
   for_each             = { for key in var.resource_keys : key.name => key }
   name                 = each.key
   resource_instance_id = ibm_resource_instance.appid.id
-  role                 = "Viewer"
+  role                 = each.value.role
+}
+
+########################################################################################################################
+# Manage authentications
+########################################################################################################################
+
+resource "ibm_appid_idp_cloud_directory" "cd" {
```

Comment:
Will all consumers always want the module to create a Management Services Cloud Directory IDP configuration? Perhaps we should have a boolean variable to let them opt in or out, and use a count around this resource?

Reply:
this can be tweaked using `is_idp_cloud_directory_active` variable

Reply:
We need at least one authentication method. And @vburckhardt said we need to have CD.

Reply:
Also, its not like CD resource is creating something, its enabling and configuring the already existing CD resource in the AppID instance. See this image:

![image](https://github.com/terraform-ibm-modules/terraform-ibm-appid/assets/108140212/2f905120-92cb-4d9e-a857-312d16f5e8c6)